### PR TITLE
Do not try to import IPython.kernel for ipython >= 4

### PR DESCRIPTION
**Pull request by larrybradley**
As reported by @keflavich (https://github.com/astropy/astropy/pull/4078#issuecomment-151397923), if `ipython` is installed but `ipykernel` is not installed then importing `astropy` from the `ipython` console gives a `IPython.kernel` deprecation warning (see #4078).  `ipykernel` is a required dependency for the `jupyter` notebook, but not for the `ipython` console.

In this PR an import of `IPython.kernel` will not be tried for `ipython >= 4`.  As a side effect, note that `OutStream` will now be `None` if using `ipython >= 4` and `ipykernel` is not installed.

### CODE REVIEW

**Comment by astrofrog** [comment]
@larrybradley - looks good, thanks! Can you add a changelog entry?

**Comment by larrybradley** [comment]
@astrofrog Done!

File: CHANGES.rst [hunk 1]
```diff
@@ -663,7 +663,7 @@ Bug fixes
 
 - ``astropy.constants``
 
-  - The constants ``Ry`` and ``u`` are now properly used inside the 
+  - The constants ``Ry`` and ``u`` are now properly used inside the
```
Comment (embray) [inline CHANGES.rst]: Can you please update this to remove the extraneous whitespace change in this commit?  It makes merging difficult.

Comment (larrybradley) [inline CHANGES.rst]: Done (my editor did that automatically).

**Comment by embray** [comment]
Cool, thanks.  I removed the extraneous whitespace in a separate commit so it won't be a problem again.

**Pull request merged**